cradle-core/src/main/java/com/exactpro/cradle/messages/MessageToStore.java

```diff
@@ -30,17 +30,23 @@ public class MessageToStore implements CradleMessage {
     private final String protocol;
     private final byte[] content;
     private MessageMetadata metadata;
+    private int serializedSize;
 
     MessageToStore(StoredMessageId id, String protocol, byte[] content) throws CradleStorageException {
+        this(id, protocol, content, null);
+    }
+
+    MessageToStore(StoredMessageId id, String protocol, byte[] content, MessageMetadata metadata) throws CradleStorageException {
         this.id = id;
         this.protocol = protocol;
         this.content = content;
+        this.metadata = metadata;
         MessageUtils.validateMessage(this);
+        this.serializedSize = MessagesSizeCalculator.calculateMessageSizeInBatch(this);
     }
 
     public MessageToStore(MessageToStore copyFrom) throws CradleStorageException {
-        this(copyFrom.getId(), copyFrom.getProtocol(), copyFrom.getContent());
-        this.metadata = copyFrom.getMetadata() != null ? new MessageMetadata(copyFrom.getMetadata()) : null;
+        this(copyFrom.getId(), copyFrom.getProtocol(), copyFrom.getContent(), copyFrom.getMetadata() != null ? new MessageMetadata(copyFrom.getMetadata()) : null);
```

Comment:
Shouldn't we pass the serialized size as well? Why do we need to calculate it again for a copy?